mlflow/models/__init__.py

```diff
@@ -81,6 +101,20 @@ def log(cls, artifact_path, flavor, registered_model_name=None, **kwargs):
             mlflow_model = cls(artifact_path=artifact_path, run_id=run_id)
             flavor.save_model(path=local_path, mlflow_model=mlflow_model, **kwargs)
             mlflow.tracking.fluent.log_artifacts(local_path, artifact_path)
+            try:
+                mlflow.tracking.fluent._record_logged_model(mlflow_model)
+            except MlflowException as e:
+                if e.error_code == databricks_pb2.ErrorCode.Name(
```

Comment:
Hm unfortunately the old MLflow 1.6.0 flask server doesn't return an `ENDPOINT_NOT_FOUND`, but rather an `INTERNAL_ERROR`, so we might have to just swallow `INTERNAL_ERROR`s as well. I tested by attempting to log a model using the client from this PR against the 1.6.0 server (adding a print statement here to log the error code, which you can see in the output below):

```
>>> import mlflow; import mlflow.pyfunc; mlflow.set_tracking_uri("http://localhost:5000"); mlflow.pyfunc.log_model("abc", loader_module="mlflow.pyfunc")
/Users/sid.murching/miniconda3/envs/mlflow-python3/lib/python3.6/site-packages/pyspark/cloudpickle.py:47: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
@SID got error code INTERNAL_ERROR
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sid.murching/code/mlflow/mlflow/pyfunc/__init__.py", line 689, in log_model
    registered_model_name=registered_model_name)
  File "/Users/sid.murching/code/mlflow/mlflow/models/__init__.py", line 117, in log
    raise e
  File "/Users/sid.murching/code/mlflow/mlflow/models/__init__.py", line 105, in log
    mlflow.tracking.fluent._record_logged_model(mlflow_model)
  File "/Users/sid.murching/code/mlflow/mlflow/tracking/fluent.py", line 328, in _record_logged_model
    MlflowClient()._record_logged_model(run_id, mlflow_model)
  File "/Users/sid.murching/code/mlflow/mlflow/tracking/client.py", line 269, in _record_logged_model
    self._tracking_client._record_logged_model(run_id, mlflow_model)
  File "/Users/sid.murching/code/mlflow/mlflow/tracking/_tracking_service/client.py", line 238, in _record_logged_model
    self.store.record_logged_model(run_id, mlflow_model)
  File "/Users/sid.murching/code/mlflow/mlflow/store/tracking/rest_store.py", line 246, in record_logged_model
    self._call_endpoint(LogModel, req_body)
  File "/Users/sid.murching/code/mlflow/mlflow/store/tracking/rest_store.py", line 32, in _call_endpoint
    return call_endpoint(self.get_host_creds(), endpoint, method, json_body, response_proto)
  File "/Users/sid.murching/code/mlflow/mlflow/utils/rest_utils.py", line 137, in call_endpoint
    response = verify_rest_response(response, endpoint)
  File "/Users/sid.murching/code/mlflow/mlflow/utils/rest_utils.py", line 107, in verify_rest_response
    raise MlflowException("%s. Response body: '%s'" % (base_msg, response.text))
mlflow.exceptions.MlflowException: API request to endpoint /api/2.0/mlflow/runs/log-model failed with error code 404 != 200. Response body: '<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>404 Not Found</title>
<h1>Not Found</h1>
<p>The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.</p>
'
```

Reply:
Oh, that's unfortunate, we'll have swallow up all exceptions then. Thanks for checking!